### PR TITLE
[Testing] Fixes relative path against _temp_fixture_easy_testing for multi OSs support

### PR DESCRIPTION
**Pull request by samsonasik**
Based on https://github.com/rectorphp/rector-src/pull/127#issuecomment-851388360

Closes https://github.com/rectorphp/rector-src/pull/127 
Closes https://github.com/symplify/symplify/pull/3245

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php [hunk 1]
```diff
@@ -49,10 +49,10 @@ protected function assertFilesWereAdded(array $expectedAddedFileWithContents): v
...
             $expectedFilePathWithContent = $expectedAddedFileWithContents[$key];
 
-            $this->assertSame(
-                $expectedFilePathWithContent->getFilePath(),
-                $addedFilePathWithContent->getFilePath()
-            );
+            [, $expectedFilePathWithContentFilePath] = explode('_temp_fixture_easy_testing', $expectedFilePathWithContent->getFilePath());
+            [, $addedFilePathWithContentFilePath] = explode('_temp_fixture_easy_testing', $addedFilePathWithContent->getFilePath());
+
+            $this->assertSame($expectedFilePathWithContentFilePath, $addedFilePathWithContentFilePath);
```
Comment (TomasVotruba) [inline packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php:60]: Looks good :+1: 

Could you add a comment why it's done this way? Just so in the future we know what is going on here :)

Comment (samsonasik) [reply packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php:60]: comment added https://github.com/rectorphp/rector-src/pull/129/commits/f17add75b6acb47693c04439bd7f150e2e60c10d

File: src/DependencyInjection/Loader/ConfigurableCallValuesCollectingPhpFileLoader.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ public function __construct(
      * @param mixed $resource
      * @param null|string $type
      */
-    public function load($resource, $type = null): void
+    public function load($resource, string $type = null): void
```
Comment (samsonasik) [inline src/DependencyInjection/Loader/ConfigurableCallValuesCollectingPhpFileLoader.php:27]: this update is required as [new symfony/dependency-injection release](https://github.com/symfony/dependency-injection/releases/tag/v5.3.0) which update the param type

File: src/DependencyInjection/Loader/ConfigurableCallValuesCollectingPhpFileLoader.php [hunk 3]
```diff
@@ -36,9 +36,9 @@ public function load($resource, $type = null): void
 
     public function import(
         $resource,
-        $type = null,
+        string $type = null,
         $ignoreErrors = false,
-        $sourceResource = null,
+        string $sourceResource = null,
```
Comment (samsonasik) [inline src/DependencyInjection/Loader/ConfigurableCallValuesCollectingPhpFileLoader.php:41]: this update is required as [new symfony/dependency-injection release](https://github.com/symfony/dependency-injection/releases/tag/v5.3.0) which update the param type

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php (hunk 1 above)
Comment (staabm) [reply packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php:60]: isn't the fix in https://github.com/rectorphp/rector-src/pull/120 for the same problem way more elegant?

Comment (TomasVotruba) [reply packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php:60]: I think this is not about line-endings, nor slahes in paths, but different rules for `sys_get_temp_dir()` functions

**Comment by TomasVotruba** [comment]
Thanks!

File: packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php (hunk 1 above)
Comment (staabm) [reply packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php:60]: still the changes in #120 fixed the problem at hand in 
grafik

Comment (TomasVotruba) [reply packages/Testing/PHPUnit/Behavior/MovingFilesTrait.php:60]: @staabm Both should be included :+1:

**Pull request merged**